add-exclude.sh

```diff
@@ -1,22 +1,162 @@
 #!/bin/bash
 set -euxo pipefail
 
-if [ $# != 1 ]; then
-    echo "Usage: $0 /path/to/trigger/branch"
+GLIBC_VERSION="${GLIBC_VERSION:-2.17}"
+
+cd "$(dirname "$0")"
+
+CRATE=${1?"USAGE: $0 CRATE"}
+date
+
+features="${FEATURES:-}"
+if [ -z "$features" ]; then
+    feature_flag=""
+else
+    feature_flag="--features"
+fi
+
+no_default_features=""
+if [ "${NO_DEFAULT_FEATURES:-}" = "true" ]; then
+    no_default_features='--no-default-features'
+fi
+
+# FIXME: make a signal handler that cleans this up if we exit early.
+if [ ! -d "${TEMPDIR:-}" ]; then
+    TEMPDIR="$(mktemp -d)"
+fi
+
+# see crawler policy: https://crates.io/policies
+curl_slowly() {
+    sleep 1 && curl --user-agent "cargo-quickinstall build pipeline (alsuren@gmail.com)" "$@"
+}
+
+export CARGO=cargo
+
+install_zig_cc_and_config_to_use_it() {
+    # Install cargo-zigbuild
+    #
+    # We use cargo-zigbuild instead of zig-cc for cargo-zigbuild has
+    # built-in for certain quirks when used with cargo-build.
+    pip3 install -r zigbuild-requirements.txt
+    
+    export CARGO=cargo-zigbuild
+    # Use our own pkg-config that fails for any input, since we cannot use
+    # locally installed lib in cross-compilation.
+    export PKG_CONFIG="$PWD/pkg-config-cross.sh"
+}
+
+REPO="$(./get-repo.sh)"
+
+if [ "${ALWAYS_BUILD:-}" != 1 ] && curl_slowly --fail -I --output /dev/null "${REPO}/releases/download/${CRATE}-${VERSION}/${CRATE}-${VERSION}-${TARGET}.tar.gz"; then
+    echo "${CRATE}/${VERSION}/${CRATE}-${VERSION}-${TARGET}.tar.gz already uploaded. Skipping."
+    exit 0
+fi
+
+# Install llvm
+if [ "${RUNNER_OS?}" == "Windows" ]; then
+    choco install llvm
+    elif [ "${RUNNER_OS?}" == "Linux" ]; then
```

Comment:
Nit: extra indent for this elif


```suggestion  
elif [ "${RUNNER_OS?}" == "Linux" ]; then
```